lkmpg.tex

```diff
@@ -1897,6 +1938,23 @@ \section{Virtual Input Device Driver}
 vinput devices are created and destroyed using sysfs.
 And, event injection is done through a \verb|/dev| node.
 The device name will be used by the userland to export a new virtual input device.
+
+The \cpp|class_attribute| structure is similar to other attribute types we talked about in section \ref{sec:sysfs}:
+
+\begin{code}
+struct class_attribute {
+    struct attribute attr;
+    ssize_t (*show)(struct class *class, struct class_attribute *attr,
+                    char *buf);
+    ssize_t (*store)(struct class *class, struct class_attribute *attr,
+                    const char *buf, size_t count);
+};
+\end{code}
+
+In \verb|vinput.c|, the macro \cpp|CLASS_ATTR_WO(export/unexport)| will generate the \cpp|class_attribute| structures which named \verb|class_attr_export/unexport|.
```

Comment:
Same as you mentioned the `include/linux/sysfs.h`. Add [`include/linux/device.h`](https://elixir.bootlin.com/linux/latest/source/include/linux/device.h#L139) for these.

Reply:
The reason why `vinput.{ch}` do not have `linux/device.h` is because it already included at [`include/input.h`](https://elixir.bootlin.com/linux/latest/source/include/linux/input.h#L19).
You could add some description about this.